### PR TITLE
Make `str.join()`s more efficient

### PR DESCRIPTION
**Pull request by nsoranzo**
by using generator expressions and tuples instead of lists.

### CODE REVIEW

File: lib/galaxy/visualization/plugins/plugin.py [hunk 1]
```diff
@@ -71,10 +71,10 @@ def __init__(self, app, path, name, config, context=None, **kwargs):
         self.name = name
         self.config = config
         base_url = context.get('base_url', '')
-        self.base_url = '/'.join([base_url, self.name]) if base_url else self.name
+        self.base_url = '/'.join((base_url, self.name)) if base_url else self.name
```
Comment (bgruening) [inline lib/galaxy/visualization/plugins/plugin.py:74]: Maybe also remove the join in favor of a pure string concat, as you did below?

**Review by bgruening — APPROVED** [review]
Thanks, @nsoranzo! Just one small comment.

File: lib/galaxy/visualization/plugins/plugin.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/visualization/plugins/plugin.py:74]: In general I think the join is clearer when there are multiple variables to join using a separator, like in this case.
In the case below there was only one variable really, so it seemed better to just append to it using `+`.
Hopefully that makes sense?

Comment (bgruening) [reply lib/galaxy/visualization/plugins/plugin.py:74]: :+1:

**Pull request merged**